scripts/actions/mobskills/leeching_current.lua

```diff
@@ -0,0 +1,27 @@
+-----------------------------------
+-- Leeching Current
+-- Description: Steals HP with a water current from targets in an area of effect.
+-- Type: Magical (Water)
+-----------------------------------
+local mobskillObject = {}
+
+mobskillObject.onMobSkillCheck = function(target, mob, skill)
+    return 0
+end
+
+mobskillObject.onMobWeaponSkill = function(target, mob, skill)
+    -- We use a var here so that mobs can use different values
+    local base = mob:getLocalVar('leechingCurrent')
+    if base == 0 then
+        base = 1000
+    end
+
+    local damage = base / skill:getTotalTargets()
```

Comment:
Chances of this happening are probably minimal or non-existing, but...

Jus in case, we should clamp this to ensure we never divide by 0.
Something like...

`local damage = base / utils.clamp(skill:getTotalTargets(),1, 18)`

Reply:
This is the same as all of the other skills like 1000 Needles. How could the total targets be 0?

If the target dies or goes out of range then the action would be cancelled. Also if players are fighting in an alliance and then leave the alliance but still have hate then I believe they'd still be able to be targeted by this. In that case limiting the total targets to 18 would be incorrect, right?